From 0104c2ed7cd71056bd278de354e640b1286ff374 Mon Sep 17 00:00:00 2001
From: quarkysg <hhtan72@yahoo.com>
Date: Tue, 27 Jul 2021 16:24:56 +0800
Subject: [PATCH] qca-ssdk-shell: SSDK console shell

This shell is used to control various switch and NSS firmware configurations.
---
 package/qca/qca-ssdk-shell/Makefile | 48 +++++++++++++++++++++++++++++
 1 file changed, 48 insertions(+)
 create mode 100644 package/qca/qca-ssdk-shell/Makefile

diff --git a/package/qca/qca-ssdk-shell/Makefile b/package/qca/qca-ssdk-shell/Makefile
new file mode 100644
index 0000000000000..a6353c3e75727
--- /dev/null
+++ b/package/qca/qca-ssdk-shell/Makefile
@@ -0,0 +1,48 @@
+
+include $(TOPDIR)/rules.mk
+include $(INCLUDE_DIR)/kernel.mk
+
+PKG_NAME:=qca-ssdk-shell
+PKG_RELEASE:=1
+
+PKG_SOURCE_URL:=https://git.codelinaro.org/clo/qsdk/oss/ssdk-shell.git
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_VERSION:=903af84df4b5f5e07544d5dd0a4d74517f4ea167
+PKG_MIRROR_HASH:=a87378a5392e4b724172fec78f4586a7bd70a9103a28959c0118ed862c620717
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/qca-ssdk-shell
+  SECTION:=QCA
+  CATEGORY:=Utilities
+  TITLE:=Shell application for QCA SSDK
+endef
+
+
+define Package/qca-ssdk-shell/description
+  This package contains a qca-ssdk shell application for QCA chipset
+endef
+
+ifndef CONFIG_TOOLCHAIN_BIN_PATH
+CONFIG_TOOLCHAIN_BIN_PATH=$(TOOLCHAIN_DIR)/bin
+endif
+
+QCASSDK_CONFIG_OPTS+= TOOL_PATH=$(CONFIG_TOOLCHAIN_BIN_PATH) \
+                SYS_PATH=$(LINUX_DIR) \
+                TOOLPREFIX=$(TARGET_CROSS) \
+                KVER=$(LINUX_VERSION) \
+                CFLAGS="$(TARGET_CFLAGS)" \
+                LDFLAGS="$(TARGET_LDFLAGS)" \
+                ARCH=$(LINUX_KARCH)
+
+define Build/Compile
+	$(MAKE) -C $(PKG_BUILD_DIR) $(strip $(QCASSDK_CONFIG_OPTS))
+endef
+
+define Package/qca-ssdk-shell/install
+	$(INSTALL_DIR) $(1)/usr/sbin
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/build/bin/ssdk_sh $(1)/usr/sbin/
+endef
+
+
+$(eval $(call BuildPackage,qca-ssdk-shell))
